### PR TITLE
Fix unsafe use of unpack

### PR DESCRIPTION
**Pull request by antoine-levitt**
Hopefully fix https://github.com/JuliaMolSim/DFTK.jl/issues/529, CC @gkemlin

### CODE REVIEW

File: src/orbitals.jl [hunk 1]
```diff
@@ -39,14 +39,16 @@ function pack_ψ(ψ)
...
-    # We unsafe_wrap the resulting array to avoid a complicated type for ψ.
-    # The resulting array is valid as long as the original x is still in live memory.
+    # We unsafe_wrap the resulting array to avoid a complicated type for ψ.    
     map(1:length(sizes_ψ)) do ik
         unsafe_wrap(Array{complex(eltype(x))},
                     pointer(@views x[ends[ik]-lengths[ik]+1:ends[ik]]),
                     sizes_ψ[ik])
     end
 end
+unpack_ψ(x, sizes_ψ) = deepcopy(unsafe_unpack_ψ, sizes_ψ)
```
Comment (mfherbst) [inline src/orbitals.jl:54]:
```suggestion
unpack_ψ(x, sizes_ψ) = deepcopy(unsafe_unpack_ψ(x, sizes_ψ))
```

Comment (antoine-levitt) [reply src/orbitals.jl]: duh

**Comment by gkemlin** [comment]
Given the results of tests, I would go for `unsafe_unpack_ψ` here too ? https://github.com/JuliaMolSim/DFTK.jl/blob/8eadc5556bd44938baa9116b9f14fcf86facbe5f/src/scf/direct_minimization.jl#L92

I have launched the tests that were yielding the segfault, I keep you update.

**Comment by gkemlin** [comment]
Well the working example from https://github.com/JuliaMolSim/DFTK.jl/issues/529 seems to work now, without raising segfault.

**Comment by antoine-levitt** [comment]
Yeah, the previous version was just wrong, I don't know what we were thinking at the time

**Pull request merged**